frontend/packages/kubevirt-plugin/src/components/vm-details.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from 'react';
+
+import { VmDetails, getName, getNamespace, getResource } from 'kubevirt-web-ui-components';
+
+import { PodKind, k8sPatch, k8sGet, K8sResourceKindReference } from '@console/internal/module/k8s';
+import { PodModel, ServiceModel, NamespaceModel } from '@console/internal/models';
+
+import {
+  ResourceLink,
+  navFactory,
+  LoadingInline,
+  NodeLink,
+  Firehose,
+} from '@console/internal/components/utils';
+import { breadcrumbsForOwnerRefs } from '@console/internal/components/utils/breadcrumbs';
+import { DetailsPage } from '@console/internal/components/factory';
+
+import { VmKind, VmiKind } from '../types';
+import { getLoadedResource } from '../utils';
+import { VirtualMachineInstanceModel, VirtualMachineInstanceMigrationModel } from '../models';
+
+import { DASHES } from './constants';
+
+// import { VmEvents } from './vm-events';
+// import VmConsolesConnected from '../vmconsoles';
+// import { Nic } from '../nic';
+// import { Disk } from '../disk';
+// import { menuActions } from './menu-actions';
+
+export const VmDetailsFirehose = ({ obj: vm, ...rest }: { obj: VmKind }) => {
+  const { name, namespace } = vm.metadata;
+
+  const resources = [
+    getResource(VirtualMachineInstanceModel, { name, namespace, isList: false }),
```

Comment:
we could probably move this method here rather sooner than later, because it is used in many places.

- note: it builds requested resource object with reasonable defaults

Reply:
And it appeared that it can be even farther enhanced. As discussed before, it will be moved along the rest of `kubevirt-web-ui-components`.